lib/preview.js

```diff
@@ -120,46 +119,55 @@ const buildUploadFileToPreviewBufferCallback = (sessionInfo, notifyMessage) => {
       return;
     }
     const destPath = getDesignManagerPath(src, dest, filePath);
-    const fileMapperArgs = buildFileMapperArgs(sessionToken)
+    const fileMapperArgs = buildFileMapperArgs(sessionToken);
     const uploadPromise = uploadFile(portalId, sessionToken, filePath, destPath);
     triggerNotify(notify, notifyMessage, filePath, uploadPromise);
   }
 }
 
 const initialPreviewBufferUpload = async (sessionInfo, filePaths) => {
-  const { portalId, sessionToken, src, dest } = sessionInfo
+  const { portalId, sessionToken, src, dest } = sessionInfo;
   const fileMapperArgs = buildFileMapperArgs(sessionToken);
   return uploadFolder(portalId, src, dest, fileMapperArgs, {}, filePaths);
 }
 
-const startPreviewWatchingService = (sessionInfo) => {
+
+const startPreviewWatcher = async (sessionInfo) => {
   const { src } = sessionInfo;
+  let watcherIsReady = false;
 
   const watcher = chokidar.watch(src, {
     ignoreInitial: true, // makes initial addition of files not trigger the watcher
     ignored: file => shouldIgnoreFile(file),
   });
 
-
   const addFileCallback = buildUploadFileToPreviewBufferCallback(sessionInfo, 'Added');
   const changeFileCallback = buildUploadFileToPreviewBufferCallback(sessionInfo, 'Change');
   const deleteFileCallback = buildDeleteFileFromPreviewBufferCallback(sessionInfo, 'file');
   const deleteFolderCallback = buildDeleteFileFromPreviewBufferCallback(sessionInfo, 'folder');
 
   watcher.on('ready', () => {
-    logger.log(
-      `This is the message that displays when previewing begins! ･°˖✧◝(⁰▿⁰)◜✧˖°.`
-    );
+    console.log('Local file watching service has started!');
+    watcherIsReady = true;
   });
   watcher.on('add', addFileCallback);
   watcher.on('change', changeFileCallback);
   watcher.on('unlink', deleteFileCallback);
   watcher.on('unlinkDir', deleteFolderCallback);
 
+  function sleep(ms) {
+    return new Promise((resolve) => {
+      setTimeout(resolve, ms);
+    });
+  }
+  while (!watcherIsReady) {
+    await sleep(1)
+    // do nothing... surely there's a better way to do this...
+  }
```

Comment:
This solution is a little awkward... 
I originally had `watch` and the local server set up at the same time since they're async. Since the server needs to generate a https cert with `mkcert-cli` and that library does some spitting out to console I wanted to block I added that helper to since the console while it runs... but that creates a race condition as whatever the watcher is doing while that runs will get silenced as well. 
So to get rid of that I wanted to just `await` both setups and do them sync. But `chokidar` doesn't allow that - instead the watcher object gets created and you can register a `ready` callback to fire once it's done. So instead we just sleep until that event fires and we know we can move on.
Another option might be to add a hook for the rest of the server setup and just throw that into the `ready` callback. But that feels just as messy if not messier... Maybe someone else has an idea! But this works in the mean time.